assets/javascripts/discourse/services/chat.js

```diff
@@ -560,6 +561,20 @@ export default Service.extend({
         }
 
         channel.set("status", busData.status);
+
+        // it is not possible for the user to set their last read message id
+        // if the channel has been archived, because all the messages have
+        // been deleted. we don't want them seeing the blue dot anymore so
+        // just completely reset the unreads
+        if (busData.status === CHANNEL_STATUSES.archived) {
```

Comment:
Will we be able to test for this?

Reply:
Hmm I think so, we can send MessageBus messages in JS tests

Reply:
Will add one